sycl/plugins/level_zero/pi_level0.hpp

```diff
@@ -102,10 +123,18 @@ struct _pi_device : _pi_object {
   // seems better to just keep it here.
   bool IsSubDevice;
 
-  // Create a new command list for executing on this device.
-  // It's caller's responsibility to remember and destroy the created
-  // command list when no longer needed.
-  pi_result createCommandList(ze_command_list_handle_t *ze_command_list);
+  // Retrieves a command list for executing on this device along with
+  // a fence to be used in tracking the execution of this command list.
+  // If a command list has been created on this device which has
+  // completed its commands, then that command list and its associated fence
+  // will be reused. Otherwise, a new command list and fence will be created for
+  // running on this device. L0 fences are created on a L0 command queue so the
+  // caller must pass a command queue to create a new fence for the new command
+  // list if a command list/fence pair is not available. All Command Lists &
+  // associated fences are destroyed at Device Release.
+  pi_result getAvailableCommandList(ze_command_list_handle_t *ZeCommandList,
+                                    ze_command_queue_handle_t ZeCommandQueue,
+                                    ze_fence_handle_t *ZeFence);
```

Comment:
Does it really make sense to return the "fence" outside? It's only usage is to be passed back to queue's executeCommandList, right? If so, we could keep it in the pi_queue's internals, I guess.

Reply:
Since the fence will change each time I wanted to explicitly pass back the fence that was being used, having the fence set as part of the pi_queue is dangerous because that fence cannot be reused since it is associated with the specific command list execution.

Reply:
My idea was that you'd pass PI queue to `getAvailableCommandList`, have it retrieve the available command list and associated fence, remember the fence into PI queue (from where `executeCommandList` will retrieve it), and return the command list. Can it work?

Reply:
You could, that instance then would have a fence set, but would that be ok? We can't reuse that fence in the next call so why store it in a pi_queue variable? The point of using a variable for the fence is because it is only valid for the single call to queue execute. The fence becomes invalid as soon as you call execute command lists.

Reply:
I see your point, it will be a short living whose lifetime is much less than the rest of pi-queue object. OK for now.